src/Neo/SmartContract/ApplicationEngine.Runtime.cs

```diff
@@ -327,9 +327,16 @@ protected internal BigInteger GetRandom()
         /// <param name="state">The message of the log.</param>
         protected internal void RuntimeLog(byte[] state)
         {
-            if (state.Length > MaxNotificationSize) throw new ArgumentException(null, nameof(state));
-            string message = Utility.StrictUTF8.GetString(state);
-            Log?.Invoke(this, new LogEventArgs(ScriptContainer, CurrentScriptHash, message));
+            if (state.Length > MaxNotificationSize) throw new ArgumentException("Message is too long.", nameof(state));
+            try
+            {
+                string message = Utility.StrictUTF8.GetString(state);
+                Log?.Invoke(this, new LogEventArgs(ScriptContainer, CurrentScriptHash, message));
+            }
+            catch
+            {
+                throw new ArgumentException("Failed to convert byte array to string: Invalid or non-printable UTF-8 sequence detected.", nameof(state));
```

Comment:
Add the failed data in hex?

Reply:
I am not sure, cause user passes in bytes, if you show them hex, does that help at all?

Reply:
It's the message, both options are good for me